spec.html

```diff
@@ -36049,6 +36071,117 @@ <h1>
             1. Return _index_ + _cp_.[[CodeUnitCount]].
           </emu-alg>
         </emu-clause>
+
+        <emu-clause id="sec-getstringindex" type="abstract operation">
+          <h1>
+            GetStringIndex (
+              _S_: a String,
+              _Input_: a List of characters derived from _S_,
+              _e_: a non-negative integer,
+            )
+          </h1>
+          <dl class="header">
+          </dl>
+          <emu-alg>
+            1. Assert: _Input_ contains the code points of StringToCodePoints(_S_).
+            1. If _S_ is the empty String, return 0.
+            1. Let _eUTF_ be the smallest index into _S_ that corresponds to the character at element _e_ of _Input_. If _e_ is greater than or equal to the number of elements in _Input_, then _eUTF_ is the number of code units in _S_.
+            1. Return _eUTF_.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-match-records">
+          <h1>Match Records</h1>
```

Comment:
I created a separate definition primarily because the only place where term "ordered pair" introduces some kind of record-like value in the spec is specific to regular expression pattern semantics. I felt that the creation of the regexp result object was far enough away from the pattern semantics to justify using spec mechanics used throughout the rest of the specification.